### PR TITLE
User can view their profile with their username and bookmarked events

### PR DESCRIPTION
**Pull request by lymichelle21**
New Stuff:
- User can view their profile with their username and bookmarked events
- User can see bookmarked feed and click on event to see details screen
- How it works:
    - Queried user bookmarks from Parse and retrieved event ids
    - If event id in local cache, retrieved event from local cache
    - If event id not in local cache, retrieved event from API

Notes:
- Would like to have a discussion about code organization for this particular commit since I needed to nest functions in order for bookmarkAdapter.notifyDataSetChanged() to not call something null - wondering if this has to do with async requests
- Planning to populate information below username with events user is interested in, user bio, user photo, and more as a stretch goal

Screenshot:
Screen Shot 2022-07-11 at 2 08 03 PM

### CODE REVIEW

File: app/src/main/java/com/capstone/event_finder/activities/MainActivity.java [hunk 1]
```diff
@@ -64,6 +67,50 @@ private void setUpBottomNavigation(FragmentManager fragmentManager) {
...
+
+    private void formatAndSetEventLocation(JsonObject temp, Event event) {
+        JsonArray formattedLocation = temp.getAsJsonObject("location").getAsJsonArray("display_address");
+        StringBuilder formattedLocationString = new StringBuilder();
+        for (int i = 0; i < formattedLocation.size(); i++) {
+            formattedLocationString.append(formattedLocation.get(i).getAsString()).append(" ");
+        }
+        event.setLocation(formattedLocationString.toString());
+    }
+
```
Comment (lymichelle21) [inline app/src/main/java/com/capstone/event_finder/activities/MainActivity.java:107]: Ignore the extra spacing here - realized I sent out the PR before deleting the space

File: app/src/main/java/com/capstone/event_finder/fragments/ProfileFragment.java [hunk 2]
```diff
@@ -24,6 +55,102 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
+        rvBookmarks = view.findViewById(R.id.rvBookmarks);
+        tvProfileUsername.setText(ParseUser.getCurrentUser().getUsername());
+
+        JsonArray allBookmarks = new JsonArray();
+        setUpRecyclerView(view);
+        getAndSetUserBookmarks(allBookmarks);
+    }
+
+    public void getAndSetUserBookmarks(JsonArray allBookmarks) {
+        bookmarkList.clear();
```
Comment (erikweissfb) [inline app/src/main/java/com/capstone/event_finder/fragments/ProfileFragment.java:69]: ❓ Why did you decide to clear the bookmark list first?

File: app/src/main/java/com/capstone/event_finder/fragments/ProfileFragment.java [hunk 3]
```diff
@@ -24,6 +55,102 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
+        userBookmarks.clear();
+        bookmarkIds.clear();
+        queryUserBookmarksFromParse(allBookmarks);
+    }
+
+    private void tryRetrieveEventInCache(String eventId, JsonArray allBookmarks) {
```
Comment (erikweissfb) [inline app/src/main/java/com/capstone/event_finder/fragments/ProfileFragment.java:75]: Each method here on its own is good, and having this in the ProfileFragment is fine to proof of concept for the MVP.  However, in the end abstracting this into the ViewModel can help clean this up a bunch.  See my note in `EventViewModel.java`

**Review by erikweissfb — APPROVED** [review]
This is looking good, great to see progress here!  Similar notes to the previous PR, now that this is functionally working, there is an opportunity to improve the architecture to one you'd typically see in other apps.

File: app/src/main/java/com/capstone/event_finder/fragments/ProfileFragment.java (hunk 2 above)
Comment (lymichelle21) [reply app/src/main/java/com/capstone/event_finder/fragments/ProfileFragment.java:69]: By clearing the bookmark list, duplicates are prevented from being added to the feed.

**Pull request merged**